### PR TITLE
Upgrade: 1.1.2 to 1.2.0

### PR DESCRIPTION
**Pull request by bk201**
- Add helper script information.
- Add known issues.

**New known issues**
- https://github.com/harvester/harvester/issues/4246
- https://github.com/harvester/harvester/issues/3839

**Preview**: https://64e4856d660bdd05f81e24ad--harvester-preview.netlify.app/dev/upgrade/v1-1-2-to-v1-2-0

**Note**: the zh translation is kept empty now to skip the building error (missing links).

### CODE REVIEW

**Review by starbops — APPROVED** [review]
LGTM, thanks!

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+Before you start an upgrade, you can run the pre-check script to make sure the cluster is in a stable state. For more details, please visit this [URL](https://github.com/harvester/upgrade-helpers/tree/main/pre-check/v1.1.x) for the script.
+:::
+
+Once there is an upgradable version, the Harvester GUI Dashboard page will show an upgrade button. For more details, please refer to [start an upgrade](./automatic.md#start-an-upgrade).
+
+For the air-gap env upgrade, please refer to [prepare an air-gapped upgrade](./automatic.md#prepare-an-air-gapped-upgrade).
+
+
+## Known Issues
```
Comment (vickyhella) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:29]:
```suggestion
## Known issues
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+---
+
+### 1. The upgrade can't start because of `"validator.harvesterhci.io" denied the request: managed chart rancher-monitoring is not ready, please wait for it to be ready`
```
Comment (vickyhella) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:33]:
```suggestion
### 1. An upgrade can't start and reports `"validator.harvesterhci.io" denied the request: managed chart rancher-monitoring is not ready, please wait for it to be ready`
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+If a cluster is configured with a **storage network**, an upgrade can't start with the following message.
+
+![](/img/v1.2/upgrade/known_issues/3839-error.png)
+
+- Related issue
+  - [[Doc] upgrade stuck while upgrading system service with alertmanager and prometheus](https://github.com/harvester/harvester/issues/3839)
+- Workaround
```
Comment (vickyhella) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:41]:
```suggestion
- Related issue:
  - [[Doc] upgrade stuck while upgrading system service with alertmanager and prometheus](https://github.com/harvester/harvester/issues/3839)
- Workaround:
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  - https://github.com/harvester/harvester/issues/3839#issuecomment-1534438192
+
+---
+
+### 2. An upgrade is stuck on creating the upgrade repository
```
Comment (vickyhella) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:46]:
```suggestion
### 2. An upgrade is stuck in `Creating Upgrade Repository`
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+An upgrade is pending while creating upgrade the repository:
```
Comment (vickyhella) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:48]:
```suggestion
During an upgrade, **Creating Upgrade Repository** is stuck in the **Pending** state:
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+  ![](/img/v1.2/upgrade/known_issues/4246-pending.png)
+
+Please do the following operations to check if the cluster runs into the issue:
```
Comment (vickyhella) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:52]:
```suggestion
Please perform the following steps to check if the cluster runs into the issue:
```

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+1. Check the upgrade repository pod:
+
+    ![](/img/v1.2/upgrade/known_issues/4246-upgrade-repo-pod.png)
+
+    The `virt-launcher-upgrade-repo-hvst-<upgrade-name>` pod stays in `ContainerCreating`.
```
Comment (vickyhella) [inline docs/upgrade/v1-1-2-to-v1-2-0.md:58]:
```suggestion
    If the `virt-launcher-upgrade-repo-hvst-<upgrade-name>` pod stays in `ContainerCreating`, your cluster might have run into this issue. In this case, proceed with step 2.
```

**Review by vickyhella — COMMENTED** [review]
Only some nits

File: docs/upgrade/v1-1-2-to-v1-2-0.md (hunk 1 above)
Comment (bk201) [reply docs/upgrade/v1-1-2-to-v1-2-0.md]: Thank you! Fixed!

**Review by vickyhella — APPROVED** [review]

File: docs/upgrade/v1-1-2-to-v1-2-0.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+title: "Upgrade from v1.1.2 to v1.2.0"
+---
+
+<head>
+  <link rel="canonical" href="https://docs.harvesterhci.io/v1.2/upgrade/v1-1-2-to-v1-2-0"/>
+</head>
+
+:::danger
+
+Please do not upgrade a running cluster to v1.2.0 if your machine has an **Intel E810** NIC card. We saw some reports that the NIC card has a problem when added to a bonding device. Please check this issue for more info: https://github.com/harvester/harvester/issues/3860.
```
[the remark this answers is not in the document]
Comment (bk201) [reply docs/upgrade/v1-1-2-to-v1-2-0.md]: Removed the danger admonitions, e810 issue is fixed:
https://github.com/harvester/harvester/issues/3860#issuecomment-1687322977

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**